x-pack/plugin/security/qa/security-trial/build.gradle

```diff
@@ -5,7 +5,7 @@ dependencies {
   javaRestTestImplementation(testArtifact(project(xpackModule('security'))))
 }
 
-testClusters.javaRestTest {
+testClusters.matching { it.name == 'javaRestTest' }.configureEach {
```

Comment:
What's the benefit of this? Is there any notion of lazy domain objection configuration? Does this actually avoid anything?

Reply:
what happens is that we only create the cluster if it's really needed when the RestIntegTest task is actually materialised. Otherwise the test task AND the according cluster are not created at all

Reply:
Oh interesting. I assume that's just a capability of domain object containers in general then?

Reply:
yes